basculechecks/capabilitiesmap.go

```diff
@@ -72,19 +75,27 @@ func (c CapabilitiesMap) CheckAuthentication(auth bascule.Authentication, vs Par
 	// if the checker is nil, we treat it like a checker that always returns
 	// false.
 	if checker == nil {
-		return NoCapabilitiesMatch, fmt.Errorf("%w in [%v] with nil endpoint checker",
-			ErrNoValidCapabilityFound, capabilities)
+		err := errWithReason{
+			err: fmt.Errorf("%w in [%v] with nil endpoint checker",
+				ErrNoValidCapabilityFound, capabilities),
+			reason: NoCapabilitiesMatch,
+		}
+		return err
```

Comment:
Since ErrNoValidCapabilityFound is already an errWithReason, we could just return: 
```
fmt.Errorf("%w in [%v] with nil endpoint checker", ErrNoValidCapabilityFound, capabilities)
```